packages/protocol/contracts/governance/Governance.sol

```diff
@@ -649,10 +667,70 @@ contract Governance is IGovernance, Ownable, Initializable, UsingLockedGold, Ree
   }
 
   /**
-   * @notice Withdraws refunded Celo Gold commitments.
+   * @notice Whitelists the hash of a hotfix.
+   * @param txHash The hash of the hotfix to be whitelisted.
+   * @param validatorIndex The index in the validator set of the whitelisting validator.
+   */
+  function whitelist(bytes32 txHash, uint256 validatorIndex) external nonReentrant {
+    if (msg.sender == approver) {
+      // TODO: consider using more state s.t. when approver changes, approval is reverted
+      hotfixes[txHash].approved = true;
+    } else {
+      IValidators validators = IValidators(registry.getAddressForOrDie(VALIDATORS_REGISTRY_ID));
+      require(msg.sender == validators.validatorAddressFromCurrentSet(validatorIndex));
+
+      uint256 epoch = getEpochNumber(); 
+      // TODO: consider using more state s.t. recurring validators can whitelist in different epochs
+      if (epoch > hotfixes[txHash].epoch) { // also initializes when hotfixes[txHash].epoch == 0
+        hotfixes[txHash].tally = 0;
+        hotfixes[txHash].epoch = epoch;
+      }
+
+      hotfixes[txHash].tally = hotfixes[txHash].tally.add(1);
+    }
+
+    emit HotfixWhitelisted(msg.sender, txHash);
+  }
+
+  /**
+   * @notice Executes a whitelisted proposal.
+   * @param values The values of Celo Gold to be sent in the proposed transactions.
+   * @param destinations The destination addresses of the proposed transactions.
+   * @param data The concatenated data to be included in the proposed transactions.
+   * @param dataLengths The lengths of each transaction's data.
+   * @dev Reverts if the proposal is not whitelisted.
+   */
+  function hotfix(
```

Comment:
I like your alternative implementation which makes it more robust to epoch boundaries. As discussed - could sidestep gas concerns by separating `hotfix` to do the check for the current epoch and `hotfixExecute` to check that epoch and execute.